tests/ignite/distributed/comp_models/test_native.py

```diff
@@ -58,6 +58,46 @@ def test__native_dist_model_create_from_backend_bad_config():
     del os.environ["RANK"]
 
 
+@pytest.mark.distributed
+@pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip if launched as multiproc")
+def test__native_dist_model_create_from_backend_bad_slurm_config():
+    import os
+    from datetime import timedelta
+
+    os.environ["SLURM_JOB_ID"] = "1"
+
+    with pytest.raises(RuntimeError, match=r"SLURM distributed configuration is missing"):
+        _NativeDistModel.create_from_backend(backend="gloo", timeout=timedelta(seconds=10))
+
+    with pytest.raises(ValueError, match=r"Arguments rank and world_size should not be specified with SLURM"):
+        _NativeDistModel.create_from_backend(
+            backend="gloo", timeout=timedelta(seconds=10), rank=1, init_method="", world_size=1
+        )
+
+    os.environ["SLURM_PROCID"] = "0"
+    os.environ["SLURM_LOCALID"] = "0"
+    os.environ["SLURM_NTASKS"] = "1"
+    os.environ["SLURM_JOB_NODELIST"] = "localhost"
+
+    with pytest.raises(FileNotFoundError, match=r"No such file or directory: 'scontrol'"):
+        _NativeDistModel.create_from_backend(backend="gloo", timeout=timedelta(seconds=10))
+
+    os.environ["RANK"] = "1"
+
+    with pytest.raises(RuntimeError, match=r"Defined env variables"):
+        _NativeDistModel.create_from_backend(backend="gloo", timeout=timedelta(seconds=10))
+
+    del os.environ["SLURM_JOB_ID"]
+    del os.environ["SLURM_PROCID"]
+    del os.environ["SLURM_LOCALID"]
+    del os.environ["SLURM_NTASKS"]
+    del os.environ["SLURM_JOB_NODELIST"]
+    del os.environ["MASTER_PORT"]
```

Comment:
@sdesrozis maybe we have to restore the env in our code before raising issues ? We have a backup env here: https://github.com/pytorch/ignite/blob/7476dcfa870f36cf92be7c1c18804935b351ca7e/ignite/distributed/comp_models/native.py#L210

We can restore it like here : https://github.com/pytorch/ignite/blob/7476dcfa870f36cf92be7c1c18804935b351ca7e/ignite/distributed/comp_models/native.py#L296-L298

such that user wont need to reset it manually it program continues running (like here in the tests).